src/es/components/atoms/input/Input.js

```diff
@@ -0,0 +1,185 @@
+// @ts-check
+import { Shadow } from '../../web-components-toolbox/src/es/components/prototypes/Shadow.js'
+
+/**
+* @export
+* @class Input
+* @type {CustomElementConstructor}
+*/
+export default class Input extends Shadow() {
+  constructor (options = {}, ...args) {
+    super({ importMetaUrl: import.meta.url, ...options }, ...args)
+
+    this.input = this.root.querySelector('input');
+    this.textarea = this.root.querySelector('textarea');
+    this.counter = this.root.querySelector('.counter');
+    this.current = this.root.querySelector('.current');
+    this.max = this.root.querySelector('.max');
+
+    if (this.input) {
+      this.maxCharInput = this.input.getAttribute('maxlength');
+
+      if (this.max) {
+        this.max.innerText = `\u00A0/ ${this.maxCharInput}`;
+      }
+    }
+
+    if (this.textarea) {
+      this.maxCharTextarea = this.textarea.getAttribute('maxlength');
+
+      if (this.max) {
+        this.max.innerText = `\u00A0/ ${this.maxCharTextarea}`;
+      }
+    }
+
+    this.inputEventListener = event => {
+      let charCount = this.input.value.length;
+
+      if (charCount <= this.maxCharInput) {
+        this.current.innerText = charCount;
+      } else {
+        event.preventDefault();
+      }
+    }
+
+    this.textareaEventListener = event => {
+      let charCount = this.textarea.value.length;
+
+      if (charCount <= this.maxCharTextarea) {
+        this.current.innerText = charCount;
+      } else {
+        event.preventDefault();
+      }
+    }
+  }
+
+  connectedCallback () {
+    if (this.shouldRenderCSS()) this.renderCSS()
+
+    if (this.counter && this.input) {
+      this.input.addEventListener('input', this.inputEventListener);
+    }
+
+    if (this.counter && this.textarea) {
+      this.textarea.addEventListener('input', this.textareaEventListener);
+    }
+  }
+
+  disconnectedCallback () {
+    this.input.removeEventListener('input', this.inputEventListener);
```

Comment:
I would use the "if" condition to remove the event listener as well at the disconnect Callback